sdk/core/src/main/java/com/klaviyo/coresdk/networking/KlaviyoProperties.kt

```diff
@@ -0,0 +1,74 @@
+package com.klaviyo.coresdk.networking
+
+import com.klaviyo.coresdk.ConfigFileUtils
+
+// Didn't inherit from a mutablemap class cause I don't want to expose the standard put options and force this API to be the only access the user has to the map
+/**
+ * Abstract class that wraps around a map to control access to its contents.
+ * Provides helper functions to add properties to the map while controlling the keys available for entry
+ */
+abstract class KlaviyoProperties {
```

Comment:
I didn't inherit from a mutablemap class cause I don't want to expose any of the standard map functions and this way I can force this API to be the only access the user has to the contained map